chained_filter_controller/test/test_multiple_chained_filter.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright 2025 ros2_control PMC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "test_multiple_chained_filter.hpp"
+
+#include <gmock/gmock.h>
+
+#include <memory>
+#include <utility>
+
+#include "controller_interface/controller_interface.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "chained_filter_controller/chained_filter.hpp"
+
+using chained_filter_controller::ChainedFilter;
+using controller_interface::CallbackReturn;
+using testing::SizeIs;
+
+using hardware_interface::LoanedStateInterface;
+
+void MultipleChainedFilterTest::SetUpTestCase() { /*rclcpp::init(0, nullptr);*/ }
+
+void MultipleChainedFilterTest::TearDownTestCase() { /*rclcpp::shutdown(); */ }
+
+void MultipleChainedFilterTest::SetUp()
+{
+  // initialize controller
+  controller_ = std::make_unique<ChainedFilter>();
+}
+
+void MultipleChainedFilterTest::TearDown() { controller_.reset(nullptr); }
+
+void MultipleChainedFilterTest::SetUpController(
+  const std::string node_name, const std::vector<rclcpp::Parameter> & parameters)
+{
+  auto node_options = controller_->define_custom_node_options();
+  node_options.parameter_overrides(parameters);
+
```

Comment:
Don't we need to parse `multiple_chained_filter.yaml` this file here?

Reply:
this is done via the cmake method
https://github.com/ros-controls/ros2_controllers/pull/1634/files#diff-a3a22369907540a2005b908a839ba9368165749c3c9c57d6f8480a562dc2aba5R70